akka-remote/src/main/scala/akka/remote/AddressUidExtension.scala

```diff
@@ -22,15 +24,18 @@ object AddressUidExtension extends ExtensionId[AddressUidExtension] with Extensi
   override def lookup = AddressUidExtension
 
   override def createExtension(system: ExtendedActorSystem): AddressUidExtension = new AddressUidExtension(system)
+
 }
 
 class AddressUidExtension(val system: ExtendedActorSystem) extends Extension {
-  val longAddressUid: Long = {
-    // FIXME we should use a long here, but then we need to change in Cluster and RemoteWatcher also
-    //ThreadLocalRandom.current.nextLong()
-    ThreadLocalRandom.current.nextInt()
-  }
 
+  val longAddressUid: Long = ThreadLocalRandom.current.nextLong()
```

Comment:
To make it compatible I think we must use `nextInt` here if artery is *not* enabled.
Otherwise a new cluster member (running Akka 2.4.11) will generate a real long, and then exchange that with an old member (Akka 2.4.10) and then half of the bits are lost in deserialization, i.e. they will have different uids on the two nodes.